src/main/java/com/powsybl/openloadflow/network/util/GenerationActivePowerDistributionStep.java

```diff
@@ -45,7 +47,7 @@ public List<ParticipatingElement> getParticipatingElements(Collection<LfBus> bus
         return buses.stream()
                 .filter(bus -> bus.isParticipating() && !bus.isDisabled() && !bus.isFictitious())
                 .flatMap(bus -> bus.getGenerators().stream())
-                .filter(generator -> generator.isParticipating() && getParticipationFactor(generator) != 0)
+                .filter(generator -> isParticipating(generator) && getParticipationFactor(generator) != 0)
```

Comment:
we should also verify provided value is _strictly_ positive I think
```suggestion
                .filter(generator -> isParticipating(generator) && getParticipationFactor(generator) > 0)
```